File: life.pl
# life.pl -- Conway's Game of Life in Perl (version 5)

($n, $m) = (40, 80);
$g = 1000;

sub display {
   my @b = @_;
   for (my $i = 0; $i < $n; $i++) {
      for (my $j = 0; $j < $m; $j++) {
         print $b[$i][$j] ? '*' : ' '
      }
      print "\n"
   }
}

sub main {
   my @b;
   for (my $i = 0; $i < $n; $i++) {
      for (my $i = 0; $i < $n; $i++) {
         $b[$i][$j] = 0
      }
   }
   # initialization
   $b[19][41] = 1;
   $b[20][40] = 1;
   $b[21][40] = 1;
   $b[22][40] = 1;
   $b[22][41] = 1;
   $b[22][42] = 1;
   $b[22][43] = 1;
   $b[19][44] = 1;
   # end of initialization
   print "Before:\n"; display @b;

   my @nextb;
   my ($nm1, $mm1) = ($n - 1, $m - 1);
   for (my $k = 0; $k < $g; $k++) {
      for (my $i = 0; $i < $n; $i++) {
         my $up = $i != 0 ? $i - 1 : $nm1; my $down = $i != $nm1 ? $i + 1 : 0;
         for (my $j = 0; $j < $m; $j++) {
            my $left = $j != 0 ? $j - 1 : $mm1; my $right = $j != $mm1 ? $j + 1 : 0;
            my $count =
               $b[$up  ][$left ] +
               $b[$up  ][$j    ] +
               $b[$up  ][$right] +
               $b[$i   ][$right] +
               $b[$down][$right] +
               $b[$down][$j    ] +
               $b[$down][$left ] +
               $b[$i   ][$left ];
            $nextb[$i][$j] = $count == 2 && $b[$i][$j] || $count == 3
         }
      }
      { my @tmp = @b; @b = @nextb; @nextb = @tmp }
   }
   print "After $g generations:\n"; display @b
}
main;

File: life.cc
// life.cc -- Conway's Game of Life in C++ (version 2011 and later)

# include <vector>
# include <iostream>

constexpr int n = 40, m = 80;
constexpr int g = 66000;

static void display(const std::vector<std::vector<signed char>> &b) noexcept {
   for (int i = 0; i < n; ++i) {
      for (int j = 0; j < m; ++j) std::cout << (b.at(i).at(j) ? '*' : ' ');
      std::cout << '\n';
   }
}

int main() {
   auto b = std::vector<std::vector<signed char>>(n, std::vector<signed char>(m));
   // initialization
   b.at(19).at(41) =
   b.at(20).at(40) =
   b.at(21).at(40) =
   b.at(22).at(40) =
   b.at(22).at(41) =
   b.at(22).at(42) =
   b.at(22).at(43) =
   b.at(19).at(44) = 1;
   // end of initialization
   std::cout << "Before:\n"; display(b);

   auto nextb = std::vector<std::vector<signed char>>(n, std::vector<signed char>(m));
   for (int k = 0; k < g; ++k) {
      for (int i = 0; i < n; ++i) {
         constexpr int nm1 = n - 1;
         int up = i != 0 ? i - 1 : nm1, down = i != nm1 ? i + 1 : 0;
         for (int j = 0; j < m; ++j) {
            constexpr int mm1 = m - 1;
            int left = j != 0 ? j - 1 : mm1, right = j != mm1 ? j + 1 : 0;
            int count =
               b.at(up  ).at(left ) +
               b.at(up  ).at(j    ) +
               b.at(up  ).at(right) +
               b.at(i   ).at(right) +
               b.at(down).at(right) +
               b.at(down).at(j    ) +
               b.at(down).at(left ) +
               b.at(i   ).at(left );
            nextb.at(i).at(j) = count == 2 && b.at(i).at(j) || count == 3;
         }
      }
      b.swap(nextb);
   }
   std::cout << "After " << g << " generations:\n"; display(b);
   return {};
}
